pulp_2to3_migration/app/plugin/content.py

```diff
@@ -90,12 +98,22 @@ def __init__(self, migrator):
         super().__init__()
         self.migrator = migrator
 
-    async def create_artifact(self, pulp2_storage_path, expected_digests={}, expected_size=None):
+    async def create_artifact(self, pulp2_storage_path, expected_digests={}, expected_size=None,
+                              downloaded=True):
         """
         Create a hard link if possible and then create an Artifact.
 
         If it's not possible to create a hard link, file is copied to the Pulp 3 storage.
+        For non-downloaded content, artifact with its expected checksum and size is created.
         """
+        if not downloaded:
```

Comment:
If the unit has not been downloaded, then how can we create an artifact for it?  Should this be RemoteArtifact?

Reply:
It's an in-memory Artifact with digest and size (if provided) and nothing else in case of on_demand content. You need this object to pass to a DeclarativeArtifact, and later in the pipeline RemoteArtifacts will be created based on the provided info. Artifact objects won't be saved in this case.

Reply:
Ah, they aren't saved.  +1